earthpy/spatial.py

```diff
@@ -370,6 +370,107 @@ def crop_image(raster, geoms, all_touched=True):
     return out_image, out_meta
 
 
+def crop_all(
+    band_paths,
+    output_dir,
+    geoms,
+    overwrite=False,
+    all_touched=True,
+    verbose=True,
+):
+    """Takes a list of rasters and a boundary, and crops them efficiently.
+
+    Parameters
+    ----------
+    band_paths : list of file paths
+        List of paths of bands that are to be cropped.
+    output_dir : list or string
+        If a list is provided, then the output files from this function will be written to
+        each path in the list. This allows for customization of file names.
+        If a string is provided, then the output files from the function will be written
+        to that directory with the name of the original file plus a suffix of "_crop".
+    geoms : geopandas geodataframe or list of polygons
+        The spatial polygon boundaries in GeoJSON-like dict format
+        to be used to crop the image. All data outside of the polygon
+        boundaries will be set to nodata and/or removed from the image.
+    overwrite : bool (default=False)
+        Disallows files to be overwritten if they exist already. Can be changed so that files
+        can be overwritten with each run of the function.
+    all_touched : bool (default=True)
+        Include a pixel in the mask if it touches any of the
+        shapes. If False, include a pixel only if its center is within one of
+        the shapes, or if it is selected by Bresenham's line algorithm.
+        (from rasterio)
+    verbose : bool (default=True)
+        Returns a list of file paths with the files that were created in the function. Can
+        be turned off so the user doesn't have to assign the function to a variable.
+
+    Returns
+    ----------
+    return files : list
+        List of the files created by the function.
+
+    Example
+    -------
+        >>> import os
+        >>> import earthpy.spatial as es
+        >>> import geopandas as gpd
+        >>> from earthpy.io import path_to_example
+        >>> band_fnames = ["red.tif", "green.tif", "blue.tif"]
+        >>> band_paths = [path_to_example(fname) for fname in band_fnames]
+        >>> rmnp = gpd.read_file(path_to_example("rmnp.shp"))
+        >>> output_dir = os.path.commonpath(band_paths)
+        >>> output_files = es.crop_all(band_paths, output_dir, rmnp, overwrite=True)
+        >>> len(output_files)
+        3
+        >>> os.path.isfile(output_files[0])
+        True
+        >>> # Cleaning up example data
```

Comment:
Clean up example data

Reply:
i see thumbs up but i don't see the edits @nkorinek did you push changes that i missed perhaps? or not push changes?